modules/apps/foundation/captcha/captcha-api/src/main/java/com/liferay/captcha/configuration/settings/internal/CaptchaSettingsImpl.java

```diff
@@ -123,6 +128,18 @@ public boolean isSendPasswordCaptchaEnabled() {
 		return _captchaConfiguration.sendPasswordCaptchaEnabled();
 	}
 
+	@Override
+	public void setCaptchaEngine(String className) throws Exception {
+		Configuration configuration = _configurationAdmin.getConfiguration(
+			"com.liferay.captcha.configuration.CaptchaConfiguration");
+
+		Dictionary<String, Object> properties = new Hashtable<>();
+
+		properties.put("captchaEngine", className);
+
+		configuration.update(properties);
```

Comment:
you're right! this reset other properties that have already been set. i need to fix this to make sure other property values are preserved. good catch!